src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1WithdrawableTreeBuilder.kt

```diff
@@ -124,15 +127,33 @@ data class WithdrawableTreeNode(
   private fun render(depth: Int): String {
     val padding = "  " + if (depth > 0) { "-".repeat(3 * depth) + "> " } else { "" }
     val abbreviatedId = entityId.toString().substring(0, 3)
-    return padding + "$entityType($abbreviatedId), withdrawable:${status.withdrawable}, mayDirectlyWithdraw:${status.userMayDirectlyWithdraw}, ${blockingDescription()}\n" +
+    val description = "" +
+      "$entityType($abbreviatedId), " +
+      "withdrawable:${yesOrNo(status.withdrawable)}, " +
+      "mayDirectlyWithdraw:${yesOrNo(status.userMayDirectlyWithdraw)}, " +
+      "${blockingDescription()} ${exemptFromCascadeDescription()}"
+
+    return "$padding$description\n" +
       children.joinToString(separator = "") { it.render(depth + 1) }
   }
 
+  private fun yesOrNo(value: Boolean) = if (value) {
+    "Y"
+  } else {
+    "N"
+  }
+
   private fun blockingDescription() = if (status.blockAncestorWithdrawals) {
     "BLOCKING"
   } else if (isBlocked()) {
     "BLOCKED"
   } else {
     ""
   }
+
+  private fun exemptFromCascadeDescription() = if (status.exemptFromCascade) {
+    "EXEMPT FROM CASCADE"
```

Comment:
As (currently) the only reason to be "exempt from cascade" is by being an ad-hoc booking, would it be clearer if we reported:

```
EXEMPT FROM CASCADE (AD-HOC BOOKING)
```

In the future if there are other special cases we could pass a particular exemption reason in?

Reply:
good idea, i'll push as a separate PR